kernel/memory/mem_alloc.cpp

```diff
@@ -183,8 +185,19 @@ void kfree(void* ptr) {
   }
 
   if (block->previous != nullptr && (block->previous)->is_free) {
-    merge_block(block->previous, block);
     // LOG_INFO("Merging to left");
+    if (!((block->previous)->next == block)) {
```

Comment:
What is going on here ? :joy:

Reply:
Some things are better left unanswered...
@emile-s